### ISSUE TITLE
Food for thought - move css/js from FO into engine

### ISSUE THREAD

**Issue by woodpigeon** [EnvironmentAgency/flood-risk-engine#155]
(Not suggesting we tackle this before release but could be a good idea in future services).

I wonder if we have missed a trick by no treating the engine as a _complete_ app containing all its govuk styling and ancillary pages - and making the FO just a very dumb shell.

_Advantages_
- we would see the correct styling in engine development, giving us confidence in the UI without having to spin up the FO to check layout etc
- we could test govuk javascript behaviour in engine development
- would help us spot UI issues earlier
- can mount the engine at / not /fre which would be nice
- BO users would see the styled journey when they edit/create an enrollment

_Disavantages_
- In the BO, we'd need to hook into the govuk_template in the right place to add the black nav bar which identifies the page as being part of the BO, and get around any navbar styling mismatches (ie navbar will be bootstrap until they go into an enrollment, then it will be govuk css..)
- There would be a few technical issues to overcome as I seem to remember an engine (eg fre) isn't great at pulling in another engine's (eg govuk_elements_rails) asset locations in into the asset_path to let us @import css or require js - but can be done.

**Comment by reggieb** [comment]
I like the separation of function from style. I'd like to think that we could mount multiple similar engines in a common front end app. Then rolling out changes to corporate styling could need only modification of the front end host app, rather than having to go through each engine.

The real shame is that you can't mount one Rails app within another. Then we could do without the engine all together, and just mount the front end app within the back end.

**Comment by woodpigeon** [comment]
@reggieb Yes I do see your point regarding separation of function and style. 
However, thinking about it now I am more awake, we can have both; the engine remains unstyled, but to see the it fullly styled while developing it, we would just need to add the govuk gem references to spec/dummy and pull in their css and js (as the FO currently does). That would remove the need to run the FO project to test layout etc, and we can could in theory run feature specs in the engine to exercise any govuk included javascripts

**Comment by autotelik** [comment]
>>BO users would see the styled journey when they edit/create an enrollment, so they are closer to the >>assisted user's view point

My memory shot but I had a feeling from WEX that this was not required, it may even be actively not wanted. 

I don't really have an axe to grind either way for proposal, but seems to require technical hacks in BO, for no gains since personally I prefer working with the real thing, rather than dummy app anyway. Feels cleaner and for me the dummy app is for rspec testing, and for testing an example frontend in real engines i.e real gem shared amongst many apps.

Our engine is not  like this, its never gonne be used anywhere but in the FO/BO anyway
